controllers/nexus/resource/validation/validation.go

```diff
@@ -80,7 +80,21 @@ func (v *Validator) SetDefaultsAndValidate(nexus *v1alpha1.Nexus) (*v1alpha1.Nex
 }
 
 func (v *Validator) validate(nexus *v1alpha1.Nexus) error {
-	return v.validateNetworking(nexus)
+	validators := []func(*v1alpha1.Nexus) error{v.validateDeployment, v.validateNetworking}
+	for _, v := range validators {
+		if err := v(nexus); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (v *Validator) validateDeployment(nexus *v1alpha1.Nexus) error {
+	if nexus.Spec.Replicas > 1 {
+		v.log.Warn("Nexus server only supports 1 replica.", "Desired Replicas", nexus.Spec.Replicas)
+		nexus.Spec.Replicas = ensureMaximum(nexus.Spec.Replicas, 1)
+	}
```

Comment:
This should be in the default setting logic IMO. Validation shouldn't modify the resource. This is also aligned with how admission webhooks are implemented, where there is a explicit separation between the mutating webhook (responsible for default-setting and any other changes made to the CR) and the validating webhook (makes no modification and simply returns an error if needed).

Wdyt?

Reply:
Yeah, I agree. IDK why I've added in there. I'll chang it.

Reply:
I agree it's not a default, but it's a mutation of the CR. That's the main reason why I don't believe we should keep it in validation. The way it is, cohesion is reduced as the validation method is validating AND changing state. IMO, we should first make all mutations to the state and only then validate.

This is already how we ensure values for the probes:

https://github.com/m88i/nexus-operator/blob/126e9042352ffe299e41b3a0951ce1f1d5fb8708/controllers/nexus/resource/validation/validation.go#L167-L198

Reply:
If I change the mutation to defaults, defaults will be doing validation since I have to check if it's greater than 1 or not. There are no defaults. :)

So there's no cohesion also moving to set defaults.

Reply:
> we should first make all mutations to the state and only then validate.

This does not make sense in this scenario. We call the default first, which will have to validate the field, if it's false we mutate to a valid one. So are you suggesting to validate and mutate in default?

Reply:
When I think of validation, I think of simply checking the values and returning an error if they're not valid.

I think the main issue here is thinking of the phase where we change the CR as default-setting only. It's not. It's "make any changes necessary prior validation". I believe we have more to gain from making this separation than we have from altering state during validation.

More appropriate names for each procedure would be `mutate` and `validate`, following the webhook definitions.

["After all object modifications are complete, and after the incoming object is validated by the API server, validating admission webhooks are invoked and can reject requests to enforce custom policies."](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#what-are-admission-webhooks)

When a validation webhook is called, its response is either ok or not. You can't make changes during that, they won't be persisted by the API server. The only response which can modify incoming objects can only come from mutating webhooks:

["When allowing a request, a mutating admission webhook may optionally modify the incoming object as well. This is done using the patch and patchType fields in the response. The only currently supported patchType is JSONPatch. See JSON patch documentation for more details. For patchType: JSONPatch, the patch field contains a base64-encoded array of JSON patch operations."](https://kubernetes.io/docs/reference/access-authn-authz/extensible-admission-controllers/#response)

Even if we keep this in validation now, it will need to be moved to mutation in #174.

Reply:
+1 to renaming. I'll move to the `defaults` and let your PR handle the refactoring with the Admission controllers.